src/main/java/io/xdb/core/worker/WorkerService.java

```diff
@@ -29,10 +35,17 @@ public AsyncStateWaitUntilResponse handleAsyncStateWaitUntil(final AsyncStateWai
             .getObjectEncoder()
             .decode(request.getStateInput(), state.getInputType());
 
-        // TODO
-        final CommandRequest commandRequest = state.waitUntil(input);
+        final Communication communication = new Communication(workerServiceOptions.getObjectEncoder());
 
-        return new AsyncStateWaitUntilResponse().commandRequest(commandRequest);
+        final io.xdb.core.command.CommandRequest commandRequest = state.waitUntil(
+            request.getContext(),
```

Comment:
I am pretty sure we need to wrap this context as well for user experience later. 
For now to make this PR more focus, it may be better to create this wrapper first but not implementing it. Because doing this later will have to change a lot of code(testing code) as a breaking changes

Reply:
We can always modify the codes before relesing. And as I mentioned [here](https://github.com/xdblab/xdb-java-sdk/pull/9#issuecomment-1789015146), we can do it in the future.